src/middlewares/rediscache.ts

```diff
@@ -0,0 +1,32 @@
+import { NextFunction, Request, Response } from "express";
+import { getRedisClient } from "@utils/redisclient";
+
+const ttl = 60 * 24 * 30;
+export default async function cacheEndpoint(req: Request, res: Response, next: NextFunction) {
+    const endpoint = `${req.method} - ${req.originalUrl}`;
+
+    if (req.method !== "GET") {
+        console.log(`Cache middleware bypassed for non-GET request: ${endpoint}`);
+        return next();
+    }
+
+    const redisClient = await getRedisClient();
+    const cachedResponse = await redisClient?.get(req.originalUrl);
+
+    if (!cachedResponse) {
+        console.log(`No cache found for endpoint: ${endpoint}. Fetching data from database.`);
+        next(); // Get data from the database
+
+        if (res.locals.studyPlans) {
+            const value = JSON.stringify(res.locals.studyPlans);
+            if (typeof req.originalUrl === 'string' && typeof value === 'string') {
```

Comment:
`req.originalUrl` can be guaranteed to be of type string

`JSON.stringify()`  as its name, is guaranteed to convert your value to be of type string 

There is no need to check for these conditions na.

https://expressjs.com/en/4x/api.html#req.originalUrl
https://www.w3schools.com/js/js_json_stringify.asp